scripts/obtain/install.sh

```diff
@@ -123,196 +130,466 @@ check_pre_reqs() {
+# args:
+# root_path - $1
+# child_path - $2
+combine_paths() {
+    eval $invocation
+    
+    # TODO: Consider making it work with any number of paths. For now:
+    if [ ! -z "${3:-}" ]; then
+        say_err "combine_paths: Function takes two parameters."
         return 1
     fi
+    
+    local root_path=$(remove_trailing_backslash $1)
+    local child_path=$(remove_beginning_backslash $2)
+    say_verbose "combine_paths: root_path=$root_path"
+    say_verbose "combine_paths: child_path=$child_path"
+    echo "$root_path/$child_path"
+    return 0
+}
 
-    say "Preparing to install .NET Tools from '$CHANNEL' channel to '$PREFIX'"
-
-    if [ -e "$PREFIX/share/dotnet/cli/dotnet" ] && [ ! -w "$PREFIX/share/dotnet/cli/dotnet" ]; then
-        say_err "dotnet cli is already installed and not writeable. Use 'curl -sSL <url> | sudo sh' to force install."
-        say_err "If you have previously installed the cli using a package manager or installer then that is why it is write protected, and you need to run sudo to install the new version."
-        say_err "Alternatively, removing the '$PREFIX/share/dotnet' directory completely before running the script will also resolve the issue."
-        return 1
-    fi
-
-    if ! check_pre_reqs; then
-        say_err "Ending install due to missing pre-reqs"
-        return 1;
-    fi
-
-    if [ "$VERSION" == "Latest" ]; then
-      local fileVersion=latest
-    else
-      local fileVersion=$VERSION
-    fi
-
-    local os=$(current_os)
-    local installLocation="$PREFIX/share/dotnet"
-    local dotnet_url="https://dotnetcli.blob.core.windows.net/dotnet/$CHANNEL/Binaries/$VERSION"
-    local dotnet_filename="dotnet-$os-x64.$fileVersion.tar.gz"
+get_machine_architecture() {
+    eval $invocation
+    
+    # Currently the only one supported
+    echo "x64"
+    return 0
+}
 
-    if [ "$RELINK" = "0" ]; then
-        if [ "$FORCE" = "0" ]; then
-            local localVersion=$(tail -n 1 "$installLocation/cli/.version" 2>/dev/null)
-            if [ "$VERSION" == "Latest" ]; then
-                # Check if we need to bother
-                local remoteData="$(curl -s https://dotnetcli.blob.core.windows.net/dotnet/$CHANNEL/dnvm/latest.$os.version)"
-                [ $? != 0 ] && say_err "Unable to determine latest version." && return 1
+# args:
+# architecture - $1
+get_cli_architecture_from_architecture() {
+    eval $invocation
+    
+    local architecture=$(to_lowercase $1)
+    case $architecture in
+        auto)
+            echo "$(get_cli_architecture_from_architecture $(get_machine_architecture))"
+            return 0
+            ;;
+        amd64|x64)
+            echo "x64"
+            return 0
+            ;;
+        x86)
+            say_err "Architecture ``x86`` currently not supported"
+            return 1
+            ;;
+    esac
+   
+    say_err "Architecture not supported. If you think this is a bug, please report it at https://github.com/dotnet/cli/issues"
```

Comment:
what do you think of printing the architecture we're detecting as a part of this message. Will help debug when they file the issue.